wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,306 @@
+                                  color=discord.Color.red(),
+                                  timestamp=datetime.datetime.now(pytz.timezone('Canada/Pacific'))
+                                  )
+            e_obj.add_field(name="Notice", value=f"**You are PERMANENTLY BANNED from\n{self.bot.guilds[0]}\n\n"
+                            "You may NOT rejoin the guild!**")
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+
+            await member.send(embed=e_obj)
+            await member.kick(reason="Not allowed back on server.")
+
+    @commands.Cog.listener(name='on_member_ban')
+    async def intercept(self, guild: discord.Guild, member: Union[discord.User, discord.Member]):
+        """Watches for a guild ban. The guild ban is undone and the user is banned via this ban system"""
+
+        # need to read the audit log to grab mod, date, and reason
+        logger.info(f"[Ban intercept()] guild ban detected and intercepted for user='{member}'")
+        try:
+            audit_ban = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+        except Exception as e:
+            logger.info(f'error while fetching ban data: {e}')
+            await self.mod_channel.send(f"Encountered following error while intercepting a ban: {e}\n" +
+                                        "**Most likely need view audit log perms.**")
+            return
+        audit_ban = audit_ban[0]
+        logger.info(f"[Ban intercept()] audit log data retrieved for intercepted ban: {audit_ban}")
+
+        # name, id, mod, mod id, date, reason
+        username = member.name + '#' + member.discriminator
+        mod = audit_ban.user.name + '#' + audit_ban.user.discriminator
+        mod_id = audit_ban.user.id
+        date = audit_ban.created_at
+        reason = audit_ban.reason if audit_ban.reason else 'No Reason Given!'
+
+        # update blacklist and db
+        self.banlist.append(member.id)
+        await BannedUsers.insert_ban(username, member.id)
+        await BanRecords.insert_record(username, member.id, mod, mod_id, date, reason)
+
+        # unban
+        await self.bot.guilds[0].unban(member)
+        logger.info(f"[Ban intercept()] ban for {username} moved into db and guild ban was removed")
+
+        # report to council
+        e_obj = discord.Embed(title="Ban Hammer Deployed",
+                              colour=discord.Color.red())
+
+        e_obj.add_field(name="Banned User", value=f"**{username}**", inline=True)
+        e_obj.add_field(name="Moderator", value=f"**{mod}**", inline=True)
+        e_obj.add_field(name="Reason", value=f"```{reason}```", inline=False)
+        e_obj.add_field(name="Notification DM", value="NOT SENT, DUE TO NO COMMON GUILD\n", inline=False)
+        e_obj.set_footer(text="Intercepted Moderator Action")
+
+        if e_obj:
+            e_obj.timestamp = date.astimezone(pytz.timezone('Canada/Pacific'))
```

Comment:
is this supposed to be the timezone for where the user is or where the bot is?

Reply:
where user is.

Reply:
then shouldn't you be using this logic rather than assuming the user is always in Pacific timezone?
https://github.com/CSSS/wall_e/blob/521900c4649fbbd674597a9df279c91c2adad6f4/wall_e/src/resources/cogs/reminders.py#L32-L43

Reply:
this just made me realize it prob doesnt matter what the tz if set to for the timestamp. it'll prob convert appropriately based on the users discord client viewing the embed. I only set it to the pacific tz cuz sfu csss exists in pst tz.

Reply:
> it'll prob convert appropriately based on the users discord client viewing the embed.

sorry?

Reply:
the embed will convert the embeds timestamp fields timezone to whatever timezone the users discord client is using.

Reply:
so it really doesnt matter what its set to.

Reply:
so then this line is unnecessary?

Reply:
yea, just tested my guess above about the client changing the timezone and i was right. so i've removed all the localization stuff.